devnet-sdk/devstack/dsl/dsl.go

```diff
@@ -1,17 +1,55 @@
 package dsl
 
 import (
+	"context"
+	"time"
+
 	"github.com/ethereum-optimism/optimism/devnet-sdk/devstack/stack"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
 )
 
+const defaultTimeout = 30 * time.Second
+
+// common provides a set of common values and methods inherited by all DSL structs.
+// These should be kept very minimal.
+// No public methods or fields should be exposed.
+// Logger is explicitly omitted as each component should have additional context applied to its logging.
+type common struct {
+	// Ctx is the context for test execution.
+	ctx context.Context
+	// T is a minimal test interface for panic-checks / assertions.
+	t stack.T
+	// Require is a helper around the above T, ready to assert against.
+	require *require.Assertions
+}
+
 type System struct {
+	common
+	log log.Logger
 	sys stack.System
 }
 
-// TODO(#15137): decide what to expose publicly in DSL
+func (s *System) Supervisor(id stack.SupervisorID) *Supervisor {
+	super := s.sys.Supervisor(id)
+	return newSupervisor(s.common, s.log.New("id", id), super)
```

Comment:
Not entirely sure what the best way to integrate with logging is here. Currently this makes the DSL log messages look like they're coming from the actual component which I think might be confusing, but not exactly sure how best to differentiate them.